pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -190,23 +201,39 @@ func (t *kprobeTracer) GetMap(name string) *ebpf.Map {
 	}
 }
 
-func (t *kprobeTracer) GetConnections(buffer []network.ConnectionStats, filter func(*network.ConnectionStats) bool) ([]network.ConnectionStats, error) {
+// RegisterConnectionsTagFunc register a callback that will be called on selected ConnectionStats from GetConnections()
+// the callback can add tag to the connections, like :
+//   callback(tags, conn) {
+//     if conn.DPort == 80 {
+//       tags.Tag(conn, "scheme:http:client")
+//     }
+//   }
+func (t *kprobeTracer) RegisterConnectionsTagFunc(callback TagFunc) {
+	t.tagFunc = callback
+}
+
+func (t *kprobeTracer) GetConnections(buffer []network.ConnectionStats, filter func(*network.ConnectionStats) bool) ([]network.ConnectionStats, network.Tags, error) {
```

Comment:
Let's drop `network.Tags` and leave the tag de-duplication as an encoding optimization. We can still do string interning in the system-probe to reduce allocations though.